src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java

```diff
@@ -406,38 +406,50 @@ public void partitionExecute(ObTableOperationResult[] results,
         List<ObTableOperationResult> subObTableOperationResults = subObTableBatchOperationResult
             .getResults();
 
-        if (subObTableOperationResults.size() < subOperations.getTableOperations().size()) {
-            // only one result when it across failed
-            // only one result when hkv puts
-            if (subObTableOperationResults.size() == 1) {
-                ObTableOperationResult subObTableOperationResult = subObTableOperationResults
+        if (returnOneResult) {
+            if (results[0] == null) {
+                results[0] = new ObTableOperationResult();
+            }
```

Comment:
need to accumulate affected_rows when the table operation spans two or more tablet

Reply:
done